server/util/qps/qps.go

```diff
@@ -31,16 +31,26 @@ type Counter struct {
 	nValidBins uint64
 	window     time.Duration
 	startOnce  sync.Once
+	ticker     <-chan time.Time
 	stop       chan struct{}
 }
 
 // NewCounter returns a QPS counter using the given duration as the averaging
 // window. The caller must call Stop() on the returned counter when it is no
 // longer needed.
 func NewCounter(window time.Duration) *Counter {
+	return new(window, nil)
+}
+
+func NewCounterForTesting(window time.Duration, ticker <-chan time.Time) *Counter {
+	return new(window, ticker)
+}
```

Comment:
nit: probably more conventional to have only a single constructor and have it accept an instance of `clockwork.Clock` rather than a ticker channel

Reply:
this feels worse for the callers than telescoping the constructors IMO, LMK what you prefer

Reply:
woops, missed this. My order of preference is something like
1. Functional options `NewCounter(window, qps.WithClock(clock))`
2. Opts struct
3. Accept all dependencies explicitly
4. Separate `ForTesting` constructor

In our codebase so far, it seems like (2) or (3) are the most conventional options